R/g.part4_extractid.R

```diff
@@ -30,57 +30,34 @@ g.part4_extractid = function(idloc, fname, dolog, sleeplogidnum, sleeplog, accid
-    accid_bu = accid
-    getLastCharacterValue = function(x) {
-      tmp = as.character(unlist(strsplit(x,"")))
-      return(tmp[length(tmp)])
-    }
-    letter = apply(as.matrix(accid), MARGIN = c(1), FUN = getLastCharacterValue)
-    for (h in 1:length(accid)) {
-      options(warn = -1)
-      numletter = as.numeric(letter[h])
-      options(warn = 0)
-      if (is.na(numletter) == TRUE) { # do not remove latest character if it is a number
-        accid[h] = as.character(unlist(strsplit(accid[h],letter[h]))[1])
-      }
-    }
-    accid = suppressWarnings(as.numeric(accid))
-    #catch for files with only id in filename and for whom the above attempt to extract the id failed:
-    if (is.na(accid) == TRUE) accid = accid_bu
-  }
   # get matching identifier from sleeplog
-  if (dolog == TRUE) {
-    accid_num = suppressWarnings(as.numeric(accid))
-    if (sleeplogidnum == FALSE) {
-      # remove spaces in ID, to ease matching, because some accelerometer brands at several spaces behind ID
-      sleeplog$ID = gsub(pattern = " ", replacement = "", x = as.character(sleeplog$ID))
-      accid = gsub(pattern = " ", replacement = "", x = as.character(accid))
-      # attempt to match
-      matching_indices_sleeplog = which(as.character(sleeplog$ID) == as.character(accid))
-      if (length(matching_indices_sleeplog) == 0) {
-        matching_indices_sleeplog_alternative = which(sleeplog$ID == accid_num)
-        if (length(matching_indices_sleeplog_alternative) > 0) {
-          warning("\nArgument sleeplogidnum is set to FALSE, but it seems the identifiers are
-                    stored as numeric values, you may want to consider changing sleeplogidnum to TRUE")
-        }
+  if (dolog == TRUE) { # since now we do not have sleeplogidnum, we only need to check the last letter of accid, if sleeplog is available
+    if (suppressWarnings(!is.na(as.numeric(sleeplog$ID[1]))) & # sleeplog id is numeric 
+        suppressWarnings(is.na(as.numeric(accid)))) {          # but accid is not
+      # remove last character (in some studies numeric id is followed by character)
+      accid_bu = accid
+      getLastCharacterValue = function(x) {
+        tmp = as.character(unlist(strsplit(x,"")))
+        return(tmp[length(tmp)])
       }
-    } else {
-      matching_indices_sleeplog = which(sleeplog$ID == accid_num)
-      if (length(matching_indices_sleeplog) == 0) {
-        matching_indices_sleeplog_alternative = which(as.character(sleeplog$ID) == as.character(accid))
-        if (length(matching_indices_sleeplog_alternative) > 0) {
-          warning("\nArgument sleeplogidnum is set to TRUE, but it seems the identifiers are
-                    stored as character values, you may want to consider changing sleeplogidnum to TRUE")
-        } else {
-          if (is.na(accid_num) == TRUE) { # format probably incorrect
-            warning(paste0("\nSleeplog id is stored as format: ", as.character(sleeplog$ID[1]), ", while
```

Comment:
Removing the checks and warnings for mismatches in ID is not ideal. I originally added them to warn the user to avoid mysterious R errors further down.

Reply:
These checks looked for inconsistencies in the definition of `sleeplogidnum` and the actual class (character/numeric) of the sleeplog$ID, that's why I removed them (now we do not have `sleeplogidnum` and sleeplog$ID is always character). 

I have now revised this function to try to match IDs in different scenarios being less dependent on the class of the ID. Also, now the sleeplog ID that was matched to the acc ID is stored in part4 reports.